src/storage/v2/vertex_accessor.cpp

```diff
@@ -268,6 +268,9 @@ Result<PropertyValue> VertexAccessor::SetProperty(PropertyId property, const Pro
   // current code always follows the logical pattern of "create a delta" and
   // "modify in-place". Additionally, the created delta will make other
   // transactions get a SERIALIZATION_ERROR.
+  if (current_value == value && !storage_->config_.salient.items.delta_on_identical_property_update) {
+    return std::move(current_value);
```

Comment:
Do you have to `std::move` here? Shouldn't RVO take care of that since `current_value` is being declared in this scope?